### PR TITLE
RDKB-56227:DNS Connectivity Check RFC Support

### PR DESCRIPTION
**Pull request by kul2241**
Reason for change: Control DNS connectivity check feature by RFC. RFC DML should be persistent on reboot and factory reset. Test Procedure:
1.)Test the RFC DML Device.DeviceInfo.X_RDKCENTRAL-COM_RFC.Feature.DNSConnectivityCheck.Enable Risks: Medium
Priority: P1
Signed-off-by: kulvendra singh <kulvendra.singh@sky.uk>

### CODE REVIEW

**Review by S-Parthiban-Selvaraj — CHANGES_REQUESTED** [review]

File: source/TR-181/middle_layer_src/wanmgr_rdkbus_utils.c [hunk 1]
```diff
@@ -66,6 +66,94 @@
...
 
+ANSC_STATUS WanMgr_SetConnectivityCheckTypeToPSM(DML_VIRTUAL_IFACE* pVirtIf, CONNECTIVITY_CHECK_TYPE type)
+{
+    int result = ANSC_STATUS_SUCCESS;
+    int retPsmSet = CCSP_SUCCESS;
+    char param_name[BUFLEN_256] = {0};
+    char param_value[BUFLEN_256] = {0};
+
+    /* Update the wan policy information in PSM */
+    memset(param_value, 0, sizeof(param_value));
```
Comment (S-Parthiban-Selvaraj) [inline source/TR-181/middle_layer_src/wanmgr_rdkbus_utils.c:77]: memset required?

File: source/TR-181/middle_layer_src/wanmgr_rdkbus_utils.c [hunk 2]
```diff
@@ -66,6 +66,94 @@
...
+    int uiLoopCount;
+
+    int TotalIfaces = WanMgr_IfaceData_GetTotalWanIface();
+    for (uiLoopCount = 0; uiLoopCount < TotalIfaces; uiLoopCount++)
+    {
+        WanMgr_Iface_Data_t*   pWanDmlIfaceData = WanMgr_GetIfaceData_locked(uiLoopCount);
+        if(pWanDmlIfaceData != NULL)
+        {
+            DML_WAN_IFACE* pWanIfaceData = &(pWanDmlIfaceData->data);
+            for(int virIf_id=0; virIf_id< pWanIfaceData->NoOfVirtIfs; virIf_id++)
```
Comment (S-Parthiban-Selvaraj) [inline source/TR-181/middle_layer_src/wanmgr_rdkbus_utils.c:103]: What is need for for loops if this is gonna return first vitual interface value of first Wan Interface

File: source/TR-181/middle_layer_src/wanmgr_dml_iface_apis.c [hunk 3]
```diff
@@ -1046,8 +1050,16 @@ BOOL WanIfCfg_SetParamBoolValue(ANSC_HANDLE hInsContext, char* ParamName, BOOL b
             }
             if (strcmp(ParamName, "EnableIPoEHealthCheck") == 0)
             {
-                pWanDmlIface->VirtIfList->EnableIPoE = bValue;
-                ret = TRUE;
+                CONNECTIVITY_CHECK_TYPE type = bValue? WAN_CONNECTIVITY_TYPE_IHC:WAN_CONNECTIVITY_TYPE_NO_CHECK;
+                if((type != pWanDmlIface->VirtIfList->IP.ConnectivityCheckType) &&
```
Comment (S-Parthiban-Selvaraj) [inline source/TR-181/middle_layer_src/wanmgr_dml_iface_apis.c:1054]: Please add a comment for the if conditions

File: source/TR-181/middle_layer_src/wanmgr_rdkbus_apis.c [hunk 4]
```diff
@@ -380,6 +367,34 @@ int get_Virtual_Interface_FromPSM(ULONG instancenum, ULONG virtInsNum ,DML_VIRTU
...
+}
+
+int get_Remote_Virtual_Interface_FromPSM(ULONG instancenum, ULONG virtInsNum ,DML_VIRTUAL_IFACE * pVirtIf)
+{
+    int retPsmGet = CCSP_SUCCESS;
+    char param_value[256];
+    char param_name[512];
+    CcspTraceInfo(("%s %d Update Wan Virtual iface Conf from PSM \n", __FUNCTION__, __LINE__));
+
+    pVirtIf->VirIfIdx = virtInsNum;
```
Comment (S-Parthiban-Selvaraj) [inline source/TR-181/middle_layer_src/wanmgr_rdkbus_apis.c:388]: Why VirIfIdx set here?
isn't it set in the Init function

**Review by S-Parthiban-Selvaraj — CHANGES_REQUESTED** [review]

File: source/TR-181/middle_layer_src/wanmgr_rdkbus_apis.c [hunk 5]
```diff
@@ -481,19 +496,6 @@ int write_Virtual_Interface_ToPSM(ULONG instancenum, ULONG virtInsNum ,DML_VIRTU
 
     CcspTraceInfo(("%s %d Entered\n", __FUNCTION__, __LINE__));
```
[the remark this answers is not in the document]
Comment (kul2241) [reply source/TR-181/middle_layer_src/wanmgr_rdkbus_apis.c:497]: the value change persist at the time of DML change.

File: source/TR-181/middle_layer_src/wanmgr_dml_iface_apis.c (hunk 3 above)
Comment (kul2241) [reply source/TR-181/middle_layer_src/wanmgr_dml_iface_apis.c:1054]: okay

File: source/TR-181/middle_layer_src/wanmgr_rdkbus_utils.c (hunk 1 above)
Comment (kul2241) [reply source/TR-181/middle_layer_src/wanmgr_rdkbus_utils.c:77]: kind of improvement, I hope assignment would work well, i will changes it.

File: source/TR-181/middle_layer_src/wanmgr_rdkbus_utils.c (hunk 2 above)
Comment (kul2241) [reply source/TR-181/middle_layer_src/wanmgr_rdkbus_utils.c:103]: yes, impact wise, it's nothing but since it's unnecessary , i will improve it.

**Review by guto86 — APPROVED** [review]

**Review by S-Parthiban-Selvaraj — APPROVED** [review]

**Pull request merged**